src/librustc_typeck/check/method/mod.rs

```diff
@@ -32,11 +32,11 @@ mod confirm;
 mod probe;
 mod suggest;
 
-pub enum MethodError {
+pub enum MethodError<'tcx> {
     // Did not find an applicable method, but we did find various
     // static methods that may apply, as well as a list of
     // not-in-scope traits which may work.
-    NoMatch(Vec<CandidateSource>, Vec<ast::DefId>, probe::Mode),
+    NoMatch(Vec<CandidateSource>, Vec<TraitRef<'tcx>>, Vec<ast::DefId>, probe::Mode),
```

Comment:
Nit: it might be worth changing this to a struct-like enum variant, or creating a separate struct like `NoMatch(NoMatchData)`.